### PR TITLE
Adding a map for last action poll response.

### PR DESCRIPTION
**Pull request by n3wscott**
I would like to be able to codify what response the last action response should be from the fake based on the last operation the fake saw.

### CODE REVIEW

File: v2/fake/fake.go [hunk 1]
```diff
@@ -172,9 +175,16 @@ func (c *FakeClient) PollLastOperation(r *v2.LastOperationRequest) (*v2.LastOper
 	c.Mutex.Lock()
 	defer c.Mutex.Unlock()
 
+	var lastActionType ActionType
```
Comment (pmorie) [inline v2/fake/fake.go]: I don't think we should make assumptions about the ordering of operations - why not do this based on the actual operation key?

Comment (n3wscott) [reply v2/fake/fake.go]: I like the idea!

Comment (n3wscott) [reply v2/fake/fake.go]: Oh so setback, OperationKey is on the response, not the request. I only have the request.

**Comment by n3wscott** [comment]
Ok updated with the idea to use operation key. I think it is better. Thanks for the suggestion.

**Comment by kibbles-n-bytes** [comment]
Overall I'm good with this. I wonder if we'd want to predefine some operations for succeeded/in progress/failed reactions, so that we wouldn't need to constantly fill the map with common desired responses.

File: v2/fake/fake_test.go [hunk 2]
```diff
@@ -341,6 +341,95 @@ func TestPollLastOperation(t *testing.T) {
...
+		if !reflect.DeepEqual(tc.err, err) {
+			t.Errorf("%v: unexpected error; expected %+v, got %+v", tc.name, tc.err, err)
+		}
+
+		actions := fakeClient.Actions()
+		if e, a := 2, len(actions); e != a {
+			t.Errorf("%v: unexpected actions; expected %v, got %v; actions = %+v", e, a, actions)
+		}
+		if e, a := fake.PollLastOperation, actions[1].Type; e != a {
+			t.Errorf("%v: unexpected action type; expected %v, got %v", e, a)
```
Comment (kibbles-n-bytes) [inline v2/fake/fake_test.go]: Needs `tc.name` as an argument

File: v2/fake/fake_test.go [hunk 3]
```diff
@@ -341,6 +341,95 @@ func TestPollLastOperation(t *testing.T) {
...
+			t.Errorf("%v: unexpected response; expected %+v, got %+v", tc.name, tc.response, response)
+		}
+
+		if !reflect.DeepEqual(tc.err, err) {
+			t.Errorf("%v: unexpected error; expected %+v, got %+v", tc.name, tc.err, err)
+		}
+
+		actions := fakeClient.Actions()
+		if e, a := 2, len(actions); e != a {
+			t.Errorf("%v: unexpected actions; expected %v, got %v; actions = %+v", e, a, actions)
```
Comment (kibbles-n-bytes) [inline v2/fake/fake_test.go]: Needs `tc.name` as an argument

**Comment by n3wscott** [comment]
Maybe that is a followup RP if people find this to be the case when using the fake client?

**Comment by kibbles-n-bytes** [comment]
Yup, that's fine by me. Noticed a bug with the errors; looks like it's not unique to just your additions, though. :frowning_face:

**Comment by n3wscott** [comment]
> tc.name
Fixing the file.

**Comment by kibbles-n-bytes** [comment]
LGTM.

**Pull request merged**